test/functions/resolution/errors/testBadCallCandidates.14-all.good

```diff
@@ -1,7 +1,7 @@
 testBadCallCandidates.chpl:14: In method 'test':
 testBadCallCandidates.chpl:24: error: unresolved call 'C.baz(42)'
-testBadCallCandidates.chpl:39: note: this candidate did not match: D.baz(x: C)
-testBadCallCandidates.chpl:24: note: because method call receiver with type 'borrowed C'
-testBadCallCandidates.chpl:32: note: is passed to formal 'this: borrowed D'
+testBadCallCandidates.chpl:68: note: this candidate did not match: baz(x: C)
+testBadCallCandidates.chpl:24: note: because call is written as a method call
+testBadCallCandidates.chpl:68: note: but candidate function is not a method
 testBadCallCandidates.chpl:24: note: other candidates are:
-testBadCallCandidates.chpl:68: note:   baz(x: C)
+testBadCallCandidates.chpl:39: note:   D.baz(x: C)
```

Comment:
Is it clear to you why the behavior changed for this case?  Given that neither candidate is a good match, I might've expected/preferred `D.baz()` to be the one that was focused on since it's a method call (albeit on an unrelated class), so has the same syntax as the call and the same number of arguments.  I suppose it could be argued that the standalone case is better since its first argument is a `C` and the first / receiver argument of a method call involving `C` is also a `C`, and maybe that's why this is happening.  But I think my instinct would still have been to prefer the D.baz() case (e.g., maybe I got my D's and C's mixed up, or a subclass/superclass relationship backwards).

Of course, D.baz() would be even more compelling to choose as the focal candidate if it took an integer as an argument.

Overall, though, I wouldn't hold up this PR over this case unless it's a trivial delta.  Specifically, I think the improvements to the other cases are worth the change here.

Reply:
Good question! The behavior change here is indeed consistent with the code change. Namely, this PR prioritizes the candidates that mismatch due to the receiver type BELOW most other mismatching kinds, such as the number of arguments, where clauses, etc. In this particular case, the non-method `baz` has a mismatch due to the number of arguments, so it ranks above `D.baz` in the "other candidates" sorting order.

The design question is: if my call `S.baz()` did not resolve, is it more likely that I meant a different receiver vs. that I meant no receiver at all? This PR implements the latter. If we feel strongly it should be the former, I suggest creating a separate issue, as there is a non-trivial implementation and design effort that would be needed.

Reply:
I think, given the syntactic differences between a method call and procedure call, the different receiver case is more likely to be what the person was thinking of, but I'm also very comfortable putting that change off.